uefi/src/table/boot.rs

```diff
@@ -717,7 +717,12 @@ impl BootServices {
     ///
     /// This function directly calls the UEFI function `EFI_BOOT_SERVICES.SignalEvent()`.
     ///
-    /// Since the UEFI function does not return any errors, this function shouldn't either.
+    /// Note: The UEFI Specification v2.9 states that this may only return `EFI_SUCCESS`, but,
+    /// depending on the UEFI implementation, it is possible that this may return an error.
+    /// More research should be done to determine if any UEFI implementations (specifically EDK2)
+    /// return error codes for this function. To be safe, ensure that error codes are handled
+    /// properly.
```

Comment:
As I see it, the idea behind the UEFI spec returning `EFI_STATUS` even from functions that are currently listed as always returning `EFI_SUCCESS` is for future compatibility; they can change the function to return an error if that becomes necessary in the future without it being a breaking API change. And similarly we return `Result` when wrapping such a function, even though right now it always returns `Ok(())`.

So my recommendation here would be to say something nice and short like: "Currently always returns ``[`Status::SUCCESS`]``." I think the fact that the function returns `Result` is enough to imply that the return value still should be handled properly.

Reply:
Cool, I'll change it real quick. Thanks for the advice!